### PR TITLE
Add support for tooltips on disabled buttons. Use this to explain a f…

### PR DESCRIPTION
**Pull request by dabreegster**
…ew pecularities of the road editor UI.

Demo:
screencast
This also includes a change from the weekend to highlight the lane card when you hover on the map. The red background may be too jarring; we could also tune that now in this PR.

### CODE REVIEW

File: widgetry/src/widgets/button.rs [hunk 1]
```diff
@@ -433,6 +442,15 @@ impl<'b, 'a: 'b, 'c> ButtonBuilder<'a, 'c> {
         self
     }
 
+    /// Set a tooltip [`Text`] to appear when hovering over the button, when the button is
+    /// disabled.
+    ///
+    /// This tooltip is only displayed when `disabled(true)` is also called.
+    pub fn disabled_tooltip(mut self, tooltip: impl Into<Text>) -> Self {
```
Comment (dabreegster) [inline widgetry/src/widgets/button.rs:449]: Not sure this API is best. I do think disabled buttons warrant a separate tooltip. I thought about making `disabled_tooltip` imply `disabled(true)` or having the API something like `.disabled(Option<Text>)`, but all of those feel more confusing

File: widgetry/src/widgets/button.rs [hunk 2]
```diff
@@ -20,6 +20,7 @@ pub struct Button {
 
     pub(crate) hotkey: Option<MultiKey>,
     tooltip: Text,
+    disabled_tooltip: Option<Text>,
```
Comment (michaelkirk) [inline widgetry/src/widgets/button.rs:23]: Using an Option makes sense, but why isn't the regular tooltip an Option? 

My instinct is that they should be consistent, but it's not a big deal.

File: widgetry/src/widgets/button.rs (hunk 1 above)
Comment (michaelkirk) [reply widgetry/src/widgets/button.rs:449]: My first impulse was to have an api like: 
```
btn_builder
    .tooltip("normal tooltip", ControlState::Default)
    .tooltip("tooltip while disabled", ControlState::Disabled)
    .disabled(app.primary.map.get_r_edit(r) == orig_road_state)
```

backed by an implementation like:
```
fn tooltip(mut self, tooltip: impl Into<Text>, forState: ControlState) {
    self.style_mut(forState).tooltip = Some(tooltip.into());
    self
}
```

But that'd be a little weird for a couple reasons... would we treat `ControlState::Default` vs. `ControlState::Hovered` the same? Or would we not use `ControlState::Default` for tooltips?

A lesser concern would be that it'd be more verbose for the default case.


So anyway, I think where you landed makes more sense.

**Review by michaelkirk — APPROVED** [review]

File: widgetry/src/widgets/button.rs (hunk 1 above)
Comment (dabreegster) [reply widgetry/src/widgets/button.rs:449]: > would we treat ControlState::Default vs. ControlState::Hovered the same? Or would we not use ControlState::Default for tooltips?

I didn't even think of it in terms of `ControlState`, but yeah, that would be confusing! Setting it for `ControlState::Default` means if the button exists, no matter where the cursor is, it's haunted by a tooltip! ;)

File: widgetry/src/widgets/button.rs (hunk 2 above)
Comment (dabreegster) [reply widgetry/src/widgets/button.rs:23]: Fixed. `Option` is more clear than checking `txt.is_empty()`

**Pull request merged**